FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/ParquetDataProcessor.cs

```diff
@@ -48,18 +49,11 @@ public sealed class ParquetDataProcessor : IColumnDataProcessor
             _defaultSchemaConverter = schemaConverterDelegate(FhirParquetSchemaConstants.DefaultSchemaProviderKey);
             _customSchemaConverter = schemaConverterDelegate(FhirParquetSchemaConstants.CustomSchemaProviderKey);
             _logger = logger;
+            _parquetConverter = new ParquetConverter();
 
-            try
-            {
-                _parquetConverterWrapper = new ParquetConverterWrapper(fhirSchemaManager.GetAllSchemas(), arrowConfiguration.Value);
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError("Create ParquetConverterWrapper failed.");
-                throw new ParquetDataProcessorException("Create ParquetConverterWrapper failed. " + ex.Message, ex);
-            }
-
-            _logger.LogInformation($"ParquetDataProcessor initialized successfully with ArrowConfiguration: {JsonConvert.SerializeObject(arrowConfiguration.Value)}.");
+            var schemaSet = fhirSchemaManager.GetAllSchemaContent();
+            _parquetConverter.InitializeSchemaSet(schemaSet);
+            _logger.LogInformation($"ParquetDataProcessor initialized successfully with {schemaSet.Count()} parquet schemas.");
```

Comment:
Is it possible that parquet converter throws an exception while initializing schema set?

Reply:
Yes. It will throw when initializing with given schema failed

Reply:
so, will the exception be thrown by native code, and be caught by .net code? where the exception will be caught? need we add unit tests to make sure it？

Reply:
The native code will not throw exception. It will return status code. Exceptions will be thrown from managed code. We have tests for [this](https://github.com/microsoft/FHIR-Analytics-Pipelines/blob/2e1c044addf629133b890c3cac6c92b77110895a/FhirToDataLake/native/parquet/csharp/test/ParquetConverterTests.cs#L26)